lib/libvfio-user.c

```diff
@@ -151,14 +150,15 @@ dev_get_caps(vfu_ctx_t *vfu_ctx, vfu_reg_info_t *vfu_reg, bool is_migr_reg,
         sparse->header.next = 0;
         sparse->nr_areas = *nr_fds = nr_mmap_areas;
 
-        mmap_areas = vfu_reg->mmap_areas;
         for (i = 0; i < nr_mmap_areas; i++) {
-            (*fds)[i] = vfu_reg->fd;
-            sparse->areas[i].offset = (__u64)mmap_areas->areas[i].iov_base;
-            sparse->areas[i].size = mmap_areas->areas[i].iov_len;
+            struct iovec *iov = &vfu_reg->mmap_areas[i];
+
             vfu_log(vfu_ctx, LOG_DEBUG, "%s: area %d %#llx-%#llx", __func__,
-                    i, sparse->areas[i].offset,
-                    sparse->areas[i].offset + sparse->areas[i].size);
+                    i, iov->iov_base, iov->iov_base + iov->iov_len);
```

Comment:
Looks like this should be iov->iov_base + iov->iov_len - 1 ?

Reply:
sure, thanks

Reply:
actually, I'm not sure it does - if iov_len is zero, the range certainly isn't iov_base to iov_base - 1

I'll add brackets to indicate the open/closed range instead